### PR TITLE
Feature/455921 unit tests

### PR DESCRIPTION
**Pull request by maximvsidorenko**
## Description / Motivation
Covered `src\lib\page-props-factory.ts` with unit-tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

### CODE REVIEW

File: samples/nextjs/src/lib/page-props-factory.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      layoutData: layoutServiceData,
+      dictionary: {},
+    };
+    spy.on(editingDataService, 'getEditingData', () => Promise.resolve(editingData));
+
+    layoutPersonalizationUtils = new LayoutPersonalizationUtils();
+
+    factory = new SitecorePagePropsFactory(
+      new ComponentPropsService(),
+      new DictionaryServiceFactory(),
```
Comment (SergeyZubatkin) [inline samples/nextjs/src/lib/page-props-factory.test.ts:57]: Do not like that we need to create classes, it is possible to mock them? May be add sinon for this purpose?

File: samples/nextjs/src/lib/page-props-factory.test.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      layoutPersonalizationUtils,
+      layoutServiceFactory,
+      editingDataService
+    );
+
+    spy.on(layoutPersonalizationUtils, 'replacePersonalizedComponentsWithLoaderComponents');
```
Comment (SergeyZubatkin) [inline samples/nextjs/src/lib/page-props-factory.test.ts:63]: Do not like strings for creating spyes, in this case we are losing type safeting of TypeScript

File: samples/nextjs/src/lib/page-props-factory.ts [hunk 3]
```diff
@@ -46,11 +47,21 @@ export class SitecorePagePropsFactory {
...
   private dictionaryService: DictionaryService;
   private layoutPersonalizationUtils: LayoutPersonalizationUtils;
-
-  constructor() {
-    this.componentPropsService = new ComponentPropsService();
+  private layoutServiceFactory: LayoutServiceFactory;
+  private editingDataService: EditingDataService;
+
+  constructor(
+    componentPropsService: ComponentPropsService,
```
Comment (SergeyZubatkin) [inline samples/nextjs/src/lib/page-props-factory.ts:54]: filed + ctor + setting in ctor could be replaced with one line
private componentPropsService: ComponentPropsService

File: samples/nextjs/src/lib/page-props-factory.test.ts (hunk 2 above)
Comment (maximvsidorenko) [reply samples/nextjs/src/lib/page-props-factory.test.ts]: Resolved

File: samples/nextjs/src/lib/page-props-factory.ts (hunk 3 above)
Comment (maximvsidorenko) [reply samples/nextjs/src/lib/page-props-factory.ts]: Resolved

File: samples/nextjs/src/lib/page-props-factory.test.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    const componentPropsService = stubObject<ComponentPropsService>(new ComponentPropsService());
+    componentPropsService.fetchServerSideComponentProps.returns(Promise.resolve({}));
+    componentPropsService.fetchStaticComponentProps.returns(Promise.resolve({}));
+
+    layoutPersonalizationUtils = createStubInstance(LayoutPersonalizationUtils);
+
+    factory = new SitecorePagePropsFactory(
+      componentPropsService,
+      dictionaryServiceFactory,
+      layoutPersonalizationUtils as any,
```
Comment (SergeyZubatkin) [inline samples/nextjs/src/lib/page-props-factory.test.ts:74]: cast to LayoutPersonalizationUtils

**Review by SergeyZubatkin — APPROVED** [review]

**Review by oleksandrkiselov — APPROVED** [review]

**Pull request merged**